Request classrooms with middleware.

simoneduca opened the pull request.

Commit:
Request classrooms with middleware.

Commit:
Merge master

Commit:
finish resolve conflicts with master

Commit:
Get token from panoptes-client.

Successfully get classerooms back

Commit:
Fix typo.

simoneduca commented:
@rogerhutchings ready for review. Note: classrooms are not displayed, but I'd like to do that in a different PR to keep 'em short and sweet.

rogerhutchings commented on src/components/container/Classrooms.jsx:
```diff
@@ -15,25 +19,14 @@ export default class Classrooms extends React.Component {
-  }
+    console.log('componentDidMount');
+    const { dispatch } = this.props;
+    console.log('THIS PROPS', this.props)
```
remove, semicolon

simoneduca replied:
@rogerhutchings I assume the comma is not a typo :trollface: ?

rogerhutchings replied:
derp. probably should just trash the line if we don't need it?

simoneduca replied:
Yep.

rogerhutchings commented on src/reducers/reducers.js:
```diff
@@ -15,17 +15,18 @@ function selectedClassroom(state = {}, action ){
+      console.log('REDUCERS: RECEIVE_CLASSROOMS')
       return Object.assign({}, state, {
         isFetching: false,
         items: action.classrooms
```
Given that there's only one store, should the property it's stored under be a bit more descriptive (like `classrooms`)?

simoneduca replied:
Not sure about this one @rogerhutchings . The context of `items` should be clear because it is within `allClassrooms`. So I don't think `items` is ambiguous. This is what I mean:
![screen shot 2016-02-11 at 10 17 33](https://cloud.githubusercontent.com/assets/8989606/12974069/04dcce62-d0a9-11e5-98fc-3334186ae08c.png)

simoneduca replied:
Actually this doesn't matter because I realised that the object structure is wrong so `items` will need replacing anyway.

rogerhutchings committed:
Refactor middleware

- Create a single point of access for actions and reducers
- Slim down a few things
- Change structure so everything is stored in the `store` dir
- Create a single config file

rogerhutchings committed:
Stick to separate actions files

rogerhutchings committed:
Remove fetch

rogerhutchings committed:
Final final folder structure

I was totes wrong, this is better. Sorry @simone

simoneduca committed:
Merge pull request #34 from zooniverse/middleware-refactor

Refactor middleware

Commit:
Fix path

shaunanoordin commented:
- Code passes sanity check,
- ...but certain new features cannot be tested.
  - Said features require a currently unavailable Education API.
  - As a result, tests will log the (non-stopping) error: `http://localhost:3001/teachers/classrooms.json Failed to load resource: net::ERR_CONNECTION_REFUSED`
- However, since (1) this is a known issue, (2) said error doesn't compromise the overall stability of the project and (3) this update is necessary to bridge development with the Education API, I'm calling this PR validated and ready to merge.

Tested on: Chrome48/FF43/Safari9 on OSX

PR Status: **Validated**
Action: **Merging**
